graphistry/compute/chain.py

```diff
@@ -13,6 +14,12 @@
 ###############################################################################
 
 
+if TYPE_CHECKING:
```

Comment:
Maybe we can encapsulate this into another module, so other parts of the code can just import the type alias from that module; something like this (in `type_aliases.py`):
```python
from typing import Type, Any
import pandas as pd

if TYPE_CHECKING:
    DataFrameT = Type[pd.DataFrame]
else:
    DataFrameT = Any
```
And in the client code we can do something like this:
```python
from type_aliases import DataFrameT
def foor(arg: DataFrameT):
```

Reply:
The underlying challenge I'm facing is how to do type checking across multiple DF libs (cudf, dask, ...) when we only have pandas installed... Maybe they have stubs?

Reply:
Maybe we can use `typing.Union`:
```python
from typing import Type, Union, Any
import pandas as pd

_DataFrameCudf = None

try:
    import cudf
    _DataFrameCudf = cudf.DataFrame
except ImportError:
    pass

if TYPE_CHECKING:
    DataFrameT = Union[pd.DataFrame]
    if _DataFrameCudf:
        DataFrameT = Union[pd.DataFrame, _DataFrameCudf]
    # ... for dask, etc.
else:
    DataFrameT = Any
```